### ISSUE TITLE
Production server very slow / not working / mysteriously failing

### ISSUE THREAD

**Issue by timothydereuse** [DDMAL/Rodan#1042]
At least since July 24th, when Martha posted this on Slack:

> something is really wrong with rodan (production). I can add jobs to a workflow, but no ports appear at all. Can someone check what is going on? I need this to work soon as I need to show a few things during the conference. Thank you!
> I cannot import a workflow either, it stays on loading forever.

Later that day, Wanyi @softcat477 found this error message when trying to deploy:

`failed to create service rodan_iipsrv: Error response from daemon: rpc error: code = DeadlineExceeded desc = context deadline exceeded` 

Right now, trying to deploy the v2.0.14 images on production (images that work on staging and locally) results in the rodan-main container crashing. I wasn't able to get a log file to see what errors are going on (should try running without restart policy or something so that it doesn't keep restarting). I did restart the whole VM just to see if that'd help.

**Comment by martha-thomae** [comment]
When we used it with Henry and Nick (from Alamire) last Thursday, it was very slow. We didn't need to create any new workflows, just run pre-existing ones, which was fine. But the slow down was very noticeable, I haven't seen it that slow in years.

**Comment by sabrina0822** [comment]
I've been looking into it with Wanyi today and yesterday.

There are docker database volumes thar are around 700GB as @kqct pointed out. I looked into it more and it seems like we are storing backups from 2022 and there are many for each month. For example:

![Screenshot 2023-08-02 at 10 00 57 AM](https://github.com/DDMAL/Rodan/assets/90227881/9d995c25-35ca-497f-89b9-3850e97c0ccf)

This would explain the incredibly large volume size and I think we should have a system that only keeps the last ~30 backups on the server. @timothydereuse 

As for the ports not being displayed error for the older version of production (v.2.0.14 --> what we've been using for the past month). I found this error in the postgres container logs.

![Screenshot 2023-08-02 at 9 56 21 AM](https://github.com/DDMAL/Rodan/assets/90227881/f612bb74-7d93-464d-aebf-4682cc249744)

**Comment by fujinaga** [comment]
We have allocations on DRAC for hard disk space so we can use that to store old backups.

**Comment by sabrina0822** [comment]
It is back up! 

We had tried a few things to resolve this so I'll just go over it in order of what we did. 
1. We had realised that ~700GB was dedicated to old backups from 2022 to 2023 and another ~400GB was dedicated for backups from 2021 to 2022. These are stored in different folders on produciton. 
2. We noticed that we had used up 1.9/2TB of allocated memory for Rodan. 
3. We had considered allocation more space from DRAC (4TB) to store the old backups but realised that it might be a bit more complicated. The backups are stored as docker volumes in `/var/lib/docker/volumes/` which made it a bit more intimidating to attach extra storage to. 
4. We tried updating the docker version to match the version in staging that worked (that did not work). 
5. There was a `clean` script in `postgres\maintenance` that hadn't been added as a cronjob. This script would delete all backups older than 30 days ensuring that we keep at least 10 backups. We ran this script in the postgres container for the backups from 2022 to 2023 after testing it a bit which removed about 675G of data.
6. Once this was done the v2.0.14 was working well with ports coming up. 
7. We updated the docker version to the latest version of produciton. After this was done the v2.0.15 worked fine on production. 

I think there might have been a problem with the docker upgrade since we were so tight on space that might explain why changing the version didn't fix things. Production also seems a bit quicker now that there is more disk availability.

Moving forward I would like to also remove the backups from 2021 to 2022 as I don't see the point in keeping them especially since they're in a different folder and aren't mounted onto the postgres container. I also want to create a cronjob for the `clean` script to insure we don't keep a ridiculously large amount of backups again. 

Additionally, I would like to look into mounting the 4TB of extra memory onto the `/var/lib/docker/volumes/rodan_resources/_data/` folder as right now it is taking up 717GB. But I'm a little nervous about doing this. Wanted approval from others first.
